drivers/tipset_message_producer.go

```diff
@@ -141,23 +133,55 @@ func (bb *BlockBuilder) WithBLSMessageAndRet(bm *types.Message, retval []byte) *
 	return bb
 }
 
-func (bb *BlockBuilder) WithSECPMessageAndCode(sm *types.SignedMessage, code exitcode.ExitCode) *BlockBuilder {
-	bb.secpMsgs = append(bb.secpMsgs, sm)
+func (bb *BlockBuilder) WithSECPMessageAndCode(bm *types.Message, code exitcode.ExitCode) *BlockBuilder {
+	secpMsg := bb.toSignedMessage(bm)
+	bb.secpMsgs = append(bb.secpMsgs, secpMsg)
 	bb.addResult(code, EmptyReturnValue)
 	return bb
 }
 
-func (bb *BlockBuilder) WithSECPMessageAndRet(sm *types.SignedMessage, retval []byte) *BlockBuilder {
-	bb.secpMsgs = append(bb.secpMsgs, sm)
+func (bb *BlockBuilder) WithSECPMessageAndRet(bm *types.Message, retval []byte) *BlockBuilder {
+	secpMsg := bb.toSignedMessage(bm)
+	bb.secpMsgs = append(bb.secpMsgs, secpMsg)
 	bb.addResult(exitcode.Ok, retval)
 	return bb
 }
 
+func (bb *BlockBuilder) WithSECPMessageOk(bm *types.Message) *BlockBuilder {
+	secpMsg := bb.toSignedMessage(bm)
+	bb.secpMsgs = append(bb.secpMsgs, secpMsg)
+	bb.addResult(exitcode.Ok, EmptyReturnValue)
+	return bb
+}
+
+func (bb *BlockBuilder) WithSECPMessageDropped(bm *types.Message) *BlockBuilder {
+	secpMsg := bb.toSignedMessage(bm)
+	bb.secpMsgs = append(bb.secpMsgs, secpMsg)
+	return bb
+}
+
 func (bb *BlockBuilder) WithTicketCount(count int64) *BlockBuilder {
 	bb.ticketCount = count
 	return bb
 }
 
+func (bb *BlockBuilder) toSignedMessage(m *types.Message) *types.SignedMessage {
+	from := m.From
+	if from.Protocol() != address.SECP256K1 {
+		bb.TD.T.Fatalf("message From field must be SECP256K1 protocol, was: %v", from.Protocol())
```

Comment:
Sorry, this restriction prevents us testing important valid cases when `From` is an ID address of an account actor. We need to be able to resolve the ID address to corresponding account.

Reply:
Would you expect that lookup be performed by inspecting the init actors state or should chain-validation keep its own mapping?

Reply:
Addressed in latest set of changes.